apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/plugin/PluginCfg.java

```diff
@@ -12,15 +13,29 @@
 public enum PluginCfg {
     INSTANCE;
 
-    private List<String> pluginClassList = new ArrayList<String>();
+    private List<PluginDefine> pluginClassList = new ArrayList<PluginDefine>();
+    protected List<String> disabledPlugins = new ArrayList<String>();
+
+    PluginCfg() {
```

Comment:
Using the contructor for initialization should very careful, if this enum is used before config initialization, the disable mechanism will fail.

So I prefer to make the initialization as a boot service, rather than this.